samcli/local/lambdafn/runtime.py

```diff
@@ -274,13 +276,34 @@ def _get_code_dir(self, code_path):
         """
 
         if code_path and os.path.isfile(code_path) and code_path.endswith(self.SUPPORTED_ARCHIVE_EXTENSIONS):
-            decompressed_dir = _unzip_file(code_path)
+            decompressed_dir: str = _unzip_file(code_path)
             self._temp_uncompressed_paths_to_be_cleaned += [decompressed_dir]
             return decompressed_dir
 
         LOG.debug("Code %s is not a zip/jar file", code_path)
         return code_path
 
+    def _unarchived_layer(self, layer: Union[str, Dict, LayerVersion]) -> Union[str, Dict, LayerVersion]:
+        """
+        If the layer's content uri points to a supported local archive file, use self._get_code_dir() to
```

Comment:
Just wanted to confirm, this only works if layer zip is present on local?

Reply:
yes, same as how we handle `Function` today

Reply:
if customers provide a S3 zip url, it is not gonna work